app/routes.py

```diff
@@ -1 +1,220 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, jsonify, make_response, abort
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+import datetime
+
+from dotenv import load_dotenv
+import requests 
+import os
+load_dotenv()
+
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+
+    if not "title" in request_body or not "description" in request_body:
+        abort(make_response({"details": "Invalid data"},400))
+
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"]
+        # completed_at = request_body["completed_at"]
+    )
+    
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_result()}, 201
+
+
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+    sort = request.args.get("sort")
+
+    if sort == "asc":
+        all_tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort == "desc":
+        all_tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        all_tasks = Task.query.all()
+
+
+    response = []
+    for task in all_tasks:
+        response.append(task.to_result())
+
+    return jsonify(response), 200 
+
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    one_task = validate_item(Task, task_id)
+
+    return make_response({"task": one_task.to_result()})
+
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_item(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return make_response({"task": task.to_result()})
```

Comment:
Note that the `make_response()` function isn't necessary here. `make_response()` should be used when we need access to the Response Object to update the headers or in situations where we need to pass the Response Object to another function like `abort()`. Otherwise, flask will handle converting our returned dictionary and status code into a proper HTTP Response Object.